### PR TITLE
Archive improved search functionality

### PR DESCRIPTION
**Pull request by r0man1337**
I removed custom filter function from VideoTable component and created two computed properties in it. Using such filtering does not change the result but allows to check if there is not empty search result and only then display second table.

Also I did little optimization of displaying tables at /videos page by replacing two created (but hidden with css) components with one component and conditional passing of parameters

archive-search

### CODE REVIEW

File: components/VideoTable.vue [hunk 1]
```diff
@@ -1,61 +1,121 @@
...
+                </v-col>
+              </v-row>
+            </td>
+          </template>
+        </v-data-table>
+        <template v-if="isSecondTableAvailable">
+          <div class="headline">
+            {{ `Videos with '${search}' in the code summary or transcription` }}
+          </div>
+          <v-data-table :items="itemsFilteredBySummary"
```
Comment (jeffreybiles) [inline components/VideoTable.vue:68]: This is pretty much the same as the first table except for some variables we're feeding in... let's put it in a component.

File: pages/videos.vue [hunk 2]
```diff
@@ -1,7 +1,11 @@
 <template>
   <div>
-    <VideoTable :videos="publishedVideos" :headers="headers" :itemsPerPage="1000" class="hidden-xs-only" />
-    <VideoTable :videos="publishedVideos" :headers="mobileHeaders" :itemsPerPage="1000" :showExpand="false" class="hidden-sm-and-up" />
+    <VideoTable
+      :videos="publishedVideos"
+      :headers="$vuetify.breakpoint.xsOnly ? mobileHeaders : headers"
```
Comment (jeffreybiles) [inline pages/videos.vue:5]: Nice improvements.

While we're there, let's make a computed property `isMobile() { return this.$vuetify.breakpoint.xsOnly }`, so it's named more succintly.

File: components/VideoTable.vue [hunk 3]
```diff
@@ -97,7 +160,15 @@ import _ from 'lodash'
...
         })
-      }
+      },
+      itemsFilteredByTitleOrDescription() {
+        let isSearchTermInNameOrDescription = (item) => RegExp(this.search, 'i').test(item.name) || RegExp(this.search, 'i').test(item.description)
+        return this.search.length > 0 ? this.mungedVideos.filter(item => isSearchTermInNameOrDescription(item)) : this.mungedVideos;
+      },
+      itemsFilteredBySummary() {
+        let isSearchTermInItemCodeSummary = (item) => RegExp(this.search, 'i').test(item.code_summary)
+        return this.mungedVideos.filter(item => isSearchTermInItemCodeSummary(item));
```
Comment (jeffreybiles) [inline components/VideoTable.vue:170]: Good work having the second table be empty when no search term is entered.

File: components/VideoTable.vue [hunk 4]
```diff
@@ -97,7 +160,15 @@ import _ from 'lodash'
             courseName,
           }
         })
-      }
+      },
+      itemsFilteredByTitleOrDescription() {
+        let isSearchTermInNameOrDescription = (item) => RegExp(this.search, 'i').test(item.name) || RegExp(this.search, 'i').test(item.description)
+        return this.search.length > 0 ? this.mungedVideos.filter(item => isSearchTermInNameOrDescription(item)) : this.mungedVideos;
```
Comment (jeffreybiles) [inline components/VideoTable.vue:166]: A possible refactor of this computed property:

```js
if(this.search.length == 0) { return this.mungedVideos }

let isSearchTermInNameOrDescription = (item) => RegExp(this.search, 'i').test(item.name) || RegExp(this.search, 'i').test(item.description)
return this.mungedVideos.filter(item => isSearchTermInNameOrDescription(item))
```

File: components/VideoTable.vue [hunk 5]
```diff
@@ -148,10 +214,13 @@ import _ from 'lodash'
 </script>
 
 <style lang="scss" scoped>
+  .search-result-tables .headline {
+    margin: 15px 0;
```
Comment (jeffreybiles) [inline components/VideoTable.vue:218]: Good improvement

**Comment by jeffreybiles** [comment]
Good work.

I appreciate 
* it seems like you got the tricky edge cases right, even in some cases I didn't specify
* you improved performance by fixing where I was accidentally calling VideoTable twice
* You included a good representative screenshot

Things to improve
* The v-data-table needs to go into a component - repeating 50 nearly-identical lines is not good.
* Two minor things (see comments):
    * improving the `itemsFilteredByTitleOrDescription` computed property
    * adding a `isMobile` computed property.

File: components/VideoTable.vue (hunk 4 above)
Comment (r0man1337) [reply components/VideoTable.vue]: Done

File: pages/videos.vue (hunk 2 above)
Comment (r0man1337) [reply pages/videos.vue]: Done

**Comment by r0man1337** [comment]
>     * The v-data-table needs to go into a component - repeating 50 nearly-identical lines is not good.

I've moved the search box and element filtering to the page component. And the VideoTable component now just only displaying the data.

**Comment by jeffreybiles** [comment]
We're moving in the right direction.

Big improvements:
* How query params are dealt with
* Less repeated code

However, it does create some new problems.

The big issue is that VideoTable is used in two other screens, so this changes functionality in other parts of the app - parts that rely on search functionality for easy use.  

I realize I did [suggest a similar refactor](https://github.com/jeffreybiles/vue-screencasts-nuxt/pull/7#discussion_r405223237) to what you did, but then I said to ignore the comment without explaining _why_.

One possible fix to this is to keep VideoTable how it is here, with just the v-data-table, but create a VideoTableSearch that contains reusable filtering and query-param code.  I don't _think_ you need to be in a Nuxt Page to access $route, right?  Then we'd replace the other two uses of VideoTable with VideoTableSearch.  Let me know what you think of this fix, or if you have a different idea that might work better.

**Comment by r0man1337** [comment]
Apparently my IDE was glitchy, I thought that this component might have been used elsewhere, and I tried to check this, but the IDE didn't show any results, and I mistakenly thought that the /videos page was the only place where the component was used.

Maybe we should add a boolean parameter to VideoTable? Like "show-search-filter" that will be "false" by default.

**Comment by jeffreybiles** [comment]
Hmm, I'm not sure that I'm conveying properly what I mean.  I think it would be quicker to show in code than in words, so I'll take this one from here and tag you on the commit(s).

**Comment by jeffreybiles** [comment]
@r0man1337 check out the solution here.

**Pull request merged**